agent/agentcontainers/containers_internal_test.go

```diff
@@ -412,6 +415,218 @@ func TestConvertDockerVolume(t *testing.T) {
+		},
+		{
+			name: "container_binds",
+			expect: []codersdk.WorkspaceAgentDevcontainer{
+				{
+					CreatedAt:    time.Date(2025, 3, 11, 17, 58, 43, 522505027, time.UTC),
+					ID:           "fdc75ebefdc0243c0fce959e7685931691ac7aede278664a0e2c23af8a1e8d6a",
+					FriendlyName: "silly_beaver",
+					Image:        "debian:bookworm",
+					Labels:       map[string]string{},
+					Running:      true,
+					Status:       "running",
+					Ports:        []codersdk.WorkspaceAgentListeningPort{},
+					Volumes: map[string]string{
+						"/tmp/test/a": "/var/coder/a",
+						"/tmp/test/b": "/var/coder/b",
+					},
+				},
+			},
+		},
+		{
+			name: "container_sameport",
+			expect: []codersdk.WorkspaceAgentDevcontainer{
+				{
+					CreatedAt:    time.Date(2025, 3, 11, 17, 56, 34, 842164541, time.UTC),
+					ID:           "4eac5ce199d27b2329d0ff0ce1a6fc595612ced48eba3669aadb6c57ebef3fa2",
+					FriendlyName: "modest_varahamihira",
+					Image:        "debian:bookworm",
+					Labels:       map[string]string{},
+					Running:      true,
+					Status:       "running",
+					Ports: []codersdk.WorkspaceAgentListeningPort{
+						{
+							Network: "tcp",
+							Port:    12345,
+						},
+					},
+					Volumes: map[string]string{},
+				},
+			},
+		},
+		{
+			name: "container_differentport",
+			expect: []codersdk.WorkspaceAgentDevcontainer{
+				{
+					CreatedAt:    time.Date(2025, 3, 11, 17, 57, 8, 862545133, time.UTC),
+					ID:           "3090de8b72b1224758a94a11b827c82ba2b09c45524f1263dc4a2d83e19625ea",
+					FriendlyName: "boring_ellis",
+					Image:        "debian:bookworm",
+					Labels:       map[string]string{},
+					Running:      true,
+					Status:       "running",
+					Ports: []codersdk.WorkspaceAgentListeningPort{
+						{
+							Network: "tcp",
+							Port:    12345,
+						},
+					},
+					Volumes: map[string]string{},
+				},
+			},
+		},
+		{
+			name: "container_volume",
+			expect: []codersdk.WorkspaceAgentDevcontainer{
+				{
+					CreatedAt:    time.Date(2025, 3, 11, 17, 59, 42, 39484134, time.UTC),
+					ID:           "b3688d98c007f53402a55e46d803f2f3ba9181d8e3f71a2eb19b392cf0377b4e",
+					FriendlyName: "upbeat_carver",
+					Image:        "debian:bookworm",
+					Labels:       map[string]string{},
+					Running:      true,
+					Status:       "running",
+					Ports:        []codersdk.WorkspaceAgentListeningPort{},
+					Volumes: map[string]string{
+						"/var/lib/docker/volumes/testvol/_data": "/testvol",
+					},
+				},
+			},
+		},
+		{
+			name: "devcontainer_simple",
+			expect: []codersdk.WorkspaceAgentDevcontainer{
+				{
+					CreatedAt:    time.Date(2025, 3, 11, 17, 1, 5, 751972661, time.UTC),
+					ID:           "0b2a9fcf5727d9562943ce47d445019f4520e37a2aa7c6d9346d01af4f4f9aed",
+					FriendlyName: "optimistic_hopper",
+					Image:        "debian:bookworm",
+					Labels: map[string]string{
+						"devcontainer.config_file": "/home/coder/src/coder/coder/agent/agentcontainers/testdata/devcontainer_simple.json",
```

Comment:
A little bit of coder inflation in this path 😂

Reply:
TBH there's probably a lot of "pruning" we can do in those fixtures, but I'd like to avoid changing them if possible :)